lib/Default/SmrPlanet.class.inc

```diff
@@ -680,7 +678,7 @@ class SmrPlanet {
 		
 		//trim buildings first
 		$this->setMaxBuildings();
```

Comment:
This trimming will be broken now because the max buildings won't be updated anymore if they've been previously fetched

Reply:
Indeed, good catch. I think the correct fix is to invalidate the attributes that are related to planet type (max buildings, type info, and menu options) in the `setTypeID` function, which is the only place that we can change the typeID.

Reply:
I should add that an even _better_ fix is probably to have a `SmrPlanetType` class that correctly handles all the attributes related to planet type (again: max buildings, type info, and menu options), and hold an instance of that class in `SmrPlanet`. Then when we need to change the typeID, we just replace this instance with a new one of a different type. But I think this is more of a second-pass enhancement, since this first pass is already pretty major.

Reply:
Yeah, having a separate class for managing the planet type would be better but I agree it's a separate pass